src/java/htsjdk/samtools/metrics/MetricsFile.java

```diff
@@ -267,10 +274,9 @@ private void printHistogram(final BufferedWriter out, final FormatUtil formatter
 
             for (final Histogram<HKEY> histo : nonEmptyHistograms) {
                 final Histogram<HKEY>.Bin bin = histo.get(key);
-                final double value = (bin == null ? 0 : bin.getValue());
-
                 out.append(SEPARATOR);
-                out.append(formatter.format(value));
+
+                if(bin != null) out.append(formatter.format(bin.getValue()));
```

Comment:
You're right, it does change the functionality and now I'm not convinced this is ideal. This is the first time we're running into this because the histogram functionality wasn't tested before. It is possible that things rely on the 0s being output into the files for unequally sized histograms.

With that in mind, maybe we can look at the test format and decide the best way to test this. The current method writes and reads back the files and compares them, which fails because the 0s don't exist in the histogram but exist after the file is written out.